### PR TITLE
TRACE-131 Update Client

### PR DESCRIPTION
**Pull request by MattyJ007**

### CODE REVIEW

File: src/middleware/customTokenAuthentication.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+import logger from 'winston'
+
+import * as client from '../model/clients'
+
+const TOKEN_PATTERN = /^ *(?:[Cc][Uu][Ss][Tt][Oo][Mm]) +([A-Za-z0-9\-._~+/]+=*) *$/
```
Comment (bradsawadye) [inline src/middleware/customTokenAuthentication.js:7]: I think we should move this constant to the constants module

Comment (MattyJ007) [reply src/middleware/customTokenAuthentication.js:7]: Why

Comment (MattyJ007) [reply src/middleware/customTokenAuthentication.js:7]: I agree with you but still, why?

File: src/middleware/customTokenAuthentication.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ctx.authenticationType = 'token'
+  } catch (error) {
+    logger.error(`Custom Token could not be verified: ${error.message}`)
+    return
+  }
+}
+
+export async function koaMiddleware(ctx, next) {
+  await authenticateToken(ctx)
+  if (ctx.authenticated != null && ctx.authenticated.clientID != null) {
```
Comment (bradsawadye) [inline src/middleware/customTokenAuthentication.js:44]: shorten this if statement. ` ctx.authenticated != null to !ctx.authenticated` same for the other condition

File: src/middleware/customTokenAuthentication.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  return client.ClientModel.findOne({ customTokenID }).then((client) => {
+    if (!client) {
+      throw new Error('Client does not exist')
+    }
+    return client
+  })
+}
+
+async function authenticateToken(ctx) {
+  if (ctx.authenticated != null) {
```
Comment (bradsawadye) [inline src/middleware/customTokenAuthentication.js:19]: the condition here can be shortened to `!ctx.authenticated` (for consistency)

File: src/middleware/customTokenAuthentication.js (hunk 1 above)
Comment (bradsawadye) [reply src/middleware/customTokenAuthentication.js:7]: Code modularity. We have a module where we put constants

**Pull request merged**